index.html

```diff
@@ -573,12 +573,61 @@ <h1>{{ { US: 'USA', Intl: 'International' }[chosen_domain] }}
   "#a7f432", "#a83731", "#af6e4d", "#ff007c", "#fcd667", "#bfafb2", "#c46210",
   "#4a646c", "#a6e7ff", "#ffd0b9", "#45a27d", "#3aa8c1"];
 
+// List of style elements we have created
+var styleNodes = [];
+
+// The current mapping from locality name to color index
+var currentMapping = {}
+
 //////////////////////////////////////////////////////////////////////
 // Dynamically create all styles, to make sure the ones from the chart
-// and from the legend match
+// and from the legend match.
 //////////////////////////////////////////////////////////////////////
-function createCssStyles() {
-  for(var i=0; i < colors.length; i++) {
+function createCssStyles(localities) {
+  // Keep track of color indices that are no longer available
+  var colorsInUse = {}
+
+  var previousMapping = currentMapping;
+  currentMapping = {}
+
+  // Pass 1: for any current locality that was in the previous list, reuse
+  // its color and mark that color index as in-use
+  for (var i = 0; i < localities.length; i++) {
+    var locality = localities[i];
+
+    if (previousMapping[locality] !== undefined) {
```

Comment:
We should also avoid colorsInUse here, I think, to avoid old mappings from a few views ago that might collide.

Reply:
Not sure I follow. `colorsInUse` is a local that's only about keeping track of color indices we use in phase 1 so we avoid them in phase 2. The list never goes across views.

Reply:
The scenario is this:
(1) step 1, NY->A,WA->B.  
(2) NY->A,IN->B
(3) WA, IN shown.  previousMappings remember WA->B and IN->B.  But to avoid a collision of two localities that are both in previousMappings, we want to end up with WA->B, IN->A.

Reply:
You're right that this sequence results in incorrect behavior, good catch! But I don't think it's for the reason you state. I just pushed a different fix for it.

> we want to end up with WA->B, IN->A

I think you have that reversed? That is the broken behavior before my fix. We want to end up with WA->A, IN->B, since IN had color B in (2) and we want to keep it stable. In other word, we only look for maximum stability from the previous view, and in some cases in means instability from two views ago (WA in (3) has a different color from in (1)).

Reply:
Right, you are correct, it should be WA->A, IN->B.  Which commit should I look at for the fix?

Reply:
> Which commit (or line of code) should I look at to understand the fix?

01e9eb869fefd3e6446ea2ab68058e6a6f2afedd

Reply:
Thanks for bearing with me - I'm not sure I get it yet.  Two questions for you:

1. Is it ever possible for `previousLocalityToColorIndex` to map two localities to the same index?  I think the answer is yes, and that's OK, right?

2. Is it ever possible for `currentLocalityToColorIndex` to map two localities to the same index?  I think the answer is also yes, (but I propose that it would be better if the answer is no).

Reply:
`previous` is made equal to `current` at the start and is never modified, so the answer must be the same for both.

And I don't think two localities can ever map to the same index, because:

- Phase 1: It's either copied from `previous` which let's assume doesn't have the issue (induction style)
- Phase 2: Or it's assigned an ever increasing index that skips over any that was assigned in Phase 1

BTW, it should be noted that color indices in phase 1 and 2 are not wrapped at 26. They can have arbitrary values and assume infinite colors. The wrapping happens later when we actually map things to css classes.